packages/seacas/libraries/ioss/src/Ioss_Region.C

```diff
@@ -308,6 +308,56 @@ namespace {
       entity->field_erase(role);
     }
   }
+
+  struct NearestGroupState
+  {
+    std::string group{"/"};
+    int nearestState{-1};
+    double nearestTime{-std::numeric_limits<double>::max()};
+  };
+
+  void get_state_time_round_down(Ioss::DatabaseIO *db, const double targetTime, NearestGroupState& loc)
+  {
```

Comment:
This function looks almost exactly like the `locate_state` function below.  except for the part where the function below calls this function...   Do we need both, or at least combine similar parts.

Reply:
See comment above

Reply:
So it appears that the difference is the monotonicity check ... it's slightly different for negative times and positive times

Reply:
It looks like lines 324 .. 335 are exactly the same as lines 348 .. 359 unless I'm missing something...

Reply:
Lines 329 and 352 are where the monotonicity checks are made ... one is a strict < and the other is <=